bundles/org.openhab.binding.sleepiq/src/main/java/org/openhab/binding/sleepiq/internal/api/impl/SleepIQImpl.java

```diff
@@ -370,9 +372,28 @@ private synchronized ContentResponse doRequest(Request request, @Nullable Map<St
             logger.trace("SleepIQ: doRequest: status={} response={}", response.getStatus(),
                     response.getContentAsString());
             return response;
-        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+        } catch (InterruptedException | TimeoutException e) {
             logger.debug("SleepIQ: doRequest: Exception message={}", e.getMessage(), e);
             throw new CommunicationException("Communication error while accessing API: " + e.getMessage());
+        } catch (ExecutionException e) {
+            logger.debug("SleepIQ: doRequest: ExecutionException message={}", e.getMessage());
+            Throwable cause = e.getCause();
+            if (cause != null && cause instanceof HttpResponseException httpResponseException) {
+                Response response = httpResponseException.getResponse();
+                if (response.getStatus() == HttpStatus.UNAUTHORIZED_401) {
+                    /*
+                     * The service may respond with HTTP code 401 without any "WWW-Authenticate"
+                     * header, violating RFC 7235. Jetty will then throw HttpResponseException.
+                     * We need to handle this in order to attempt reauthentication.
+                     */
+                    logger.debug("SleepIQ: doRequest: Unauthorized, force a new login");
+                    resetLogin();
+                    throw new CommunicationException("Unauthorized, force a new login");
+                }
+            }
+            // Want the stack trace for everything else
+            logger.debug("SleepIQ: doRequest: ", e);
```

Comment:
Already logged in line 379. If you prefer having the stack trace, that line can be removed.

Reply:
Well, I didn't want the stack trace if the exception was unauthorized. But since it's debug, it really doesn't matter. I'll remove it